src/Avalonia.Native/AvaloniaNativeGlPlatformGraphics.cs

```diff
@@ -203,6 +221,137 @@ public IGlPlatformSurfaceRenderTarget CreateGlRenderTarget(IGlContext context)
             var avnContext = (GlContext)context;
             return new GlPlatformSurfaceRenderTarget(_topLevel.CreateGlRenderTarget(avnContext.Context), avnContext);
         }
+    }
+
+    class GlExternalObjectsFeature : IGlContextExternalObjectsFeature
+    {
+        private readonly GlContext _context;
+        private readonly GlDisplay _display;
+
+        public unsafe GlExternalObjectsFeature(GlContext context, GlDisplay display)
+        {
+            _context = context;
+            _display = display;
+            ulong registryId = 0;
+            if (context.Context.GetIOKitRegistryId(&registryId) != 0)
+            {
+                // We are reversing bytes to match MoltenVK (LUID is a Vulkan term after all)
+                DeviceLuid = BitConverter.GetBytes(registryId).Reverse().ToArray();
+            }
+        }
+
+        public IReadOnlyList<string> SupportedImportableExternalImageTypes { get; } =
+            [KnownPlatformGraphicsExternalImageHandleTypes.IOSurfaceRef];
+        public IReadOnlyList<string> SupportedExportableExternalImageTypes { get; } = [];
+
+        public IReadOnlyList<string> SupportedImportableExternalSemaphoreTypes { get; } =
+            [KnownPlatformGraphicsExternalSemaphoreHandleTypes.MetalSharedEvent];
+
+        public IReadOnlyList<string> SupportedExportableExternalSemaphoreTypes { get; } = [];
+        public IReadOnlyList<PlatformGraphicsExternalImageFormat> GetSupportedFormatsForExternalMemoryType(string type)
+        {
+            return [PlatformGraphicsExternalImageFormat.B8G8R8A8UNorm];
+        }
+
+        public IGlExportableExternalImageTexture CreateImage(string type, PixelSize size, PlatformGraphicsExternalImageFormat format) => 
+            throw new NotSupportedException();
+
+        public IGlExportableExternalImageTexture CreateSemaphore(string type) => throw new NotSupportedException();
+
+        public IGlExternalImageTexture ImportImage(IPlatformHandle handle, PlatformGraphicsExternalImageProperties properties)
```

Comment:
Why don't we read these properties (format, width, height) from the surface directly?

Reply:
Those aren't always readable depending on the platform (it's an implementation of an xplat interface)